node/src/neighborhood/neighborhood_database.rs

```diff
@@ -50,15 +51,20 @@ impl NeighborhoodDatabase {
             logger: Logger::new("NeighborhoodDatabase"),
         };
 
-        let mut node_record = NodeRecord::new(
-            public_key,
+        let location = if let Some(node_addr) = neighborhood_mode.node_addr_opt() {
+            get_node_location(Some(node_addr.ip_addr))
+        } else {
+            None
+        };
+        let node_record_data = NodeRecordInputs {
             earning_wallet,
-            *neighborhood_mode.rate_pack(),
-            neighborhood_mode.accepts_connections(),
-            neighborhood_mode.routes_data(),
-            0,
-            cryptde,
-        );
+            rate_pack: *neighborhood_mode.rate_pack(),
+            accepts_connections: neighborhood_mode.accepts_connections(),
+            routes_data: neighborhood_mode.routes_data(),
+            version: 0,
+            location,
+        };
+        let mut node_record = NodeRecord::new(public_key, cryptde, node_record_data);
```

Comment:
I took public_key and cryptde from node_record_data because of their lifetime. Do you think I should put it there and create that lifetime for the whole struct?